### PR TITLE
Adding Additional endpoint

### PR DESCRIPTION
**Pull request by saumya1singh**
Hi @FroMage !

Here is the PR for #14195 - adding `additional endpoints` category in `endpoints.html`  :)

### CODE REVIEW

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {#include main}
 {#title}Quarkus REST Endpoints{/title}
 {#body}
-<header>
- <h4>REST Resources</h4>
+<br>
```
Comment (FroMage) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html:4]: In general `<br>` usage is not recommended in HTML. If you feel the margin above the `h5` is not enough you should add some CSS to this page to increase the top margin.

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html [hunk 2]
```diff
@@ -21,10 +21,14 @@ <h4>REST Resources</h4>
...
-
-<header>
-
-</header>
+<br>
+ <h5>Additional Endpoints</h5>
+ {#for additionalEndpoint in info:additionalEndpointInfo}
+ <div class="row">
+  <div class="col">
+   <code>{additionalEndpoint.endpoint}   {additionalEndpoint.description}</code>
```
Comment (FroMage) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html:29]: Could you add a link to the endpoint please? And probably use the description as the link label if there is not. If not, use the endpoint url.

**Review by FroMage — CHANGES_REQUESTED** [review]

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html (hunk 1 above)
Comment (SaumyaSingh1) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html:4]: Ok , I will do that :)

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html (hunk 2 above)
Comment (SaumyaSingh1) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html:29]: ok, I will add a link to the endpoint.

**Comment by FroMage** [comment]
Also, I noticed that you have a merge commit, so you probably merged a newer version of `master` on your branch. We try to avoid that in the Quarkus code base, and prefer rebasing instead. Once you have your stuff committed, you can update your master branch (`git checkout master; git pull`) and then rebase your work branch (`git checkout -; git rebase master`) and _in theory_ if you don't have any conflicts it should pass. If you do, let us know and we'll help resolve them.

**Comment by SaumyaSingh1** [comment]
Got it

**Comment by SaumyaSingh1** [comment]
When I tried pushing using the above `rebasing` technique, it is showing this error 

`! [rejected]              AdditionalEndpoint -> AdditionalEndpoint (non-fast-forward)
error: failed to push some refs to 'https://github.com/SaumyaSingh1/quarkus'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
`

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html (hunk 2 above)
Comment (SaumyaSingh1) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html:29]: And one more doubt I had is in `NotFoundPageDisplayableEndpointBuildItem` , we have two fields only `endpoint` and `description`. 
Umm to add a link to endpoint , like what exactly I need to put inside `href` ..

**Comment by SaumyaSingh1** [comment]
> When I tried pushing using the above `rebasing` technique, it is showing this error
> 
> `! [rejected] AdditionalEndpoint -> AdditionalEndpoint (non-fast-forward) error: failed to push some refs to 'https://github.com/SaumyaSingh1/quarkus' hint: Updates were rejected because the tip of your current branch is behind hint: its remote counterpart. Integrate the remote changes (e.g. hint: 'git pull ...') before pushing again. hint: See the 'Note about fast-forwards' in 'git push --help' for details. `

oh done, at the end on `master` branch I forgot to do `git rebase AdditionalEndpoint` !

**Comment by FroMage** [comment]
This error happens when you rebase a branch and try to push it somewhere it was already pushed before, because rebasing changes your branch, so in this case you have to `git push --force` to override the check.

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html (hunk 2 above)
Comment (SaumyaSingh1) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html:29]: > And one more doubt I had is in `NotFoundPageDisplayableEndpointBuildItem` , we have two fields only `endpoint` and `description`.
> Umm to add a link to endpoint , like what exactly I need to put inside `href` ..

@FroMage need help here

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html [hunk 3]
```diff
@@ -21,10 +28,17 @@ <h4>REST Resources</h4>
...
-
-<header>
-
-</header>
+ <h5>Additional Endpoints</h5>
+ {#for additionalEndpoint in info:additionalEndpointInfo}
+ <div class="row">
+  <div class="col">
+   {#if additionalEndpoint.description}
+   {additionalEndpoint.endpoint} &emsp; <a href="{additionalEndpoint.endpoint}">{additionalEndpoint.description}</a>
```
Comment (FroMage) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html:36]: I don't think we need to add the `{additionalEndpoint.endpoint} &emsp;` part: the link is enough.

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html [hunk 4]
```diff
@@ -21,10 +28,17 @@ <h4>REST Resources</h4>
...
+   {#else if !endpoint.description}
```
Comment (FroMage) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html:37]: A simple `{#else}` should be enough: I don't think `endpoint` exists here, and we already know there's no description due to the `{#if ...` condition.

**Review by FroMage — CHANGES_REQUESTED** [review]
Great that you figured out how to rebase, we're almost there :)

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html (hunk 3 above)
Comment (SaumyaSingh1) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/resources/dev-templates/endpoints.html:36]: Hmm that make sense :)

**Comment by FroMage** [comment]
CI failed due to an unrelated hanging job (https://github.com/quarkusio/quarkus/issues/14424), I've restarted it.

**Review by FroMage — APPROVED** [review]
This is great!

**Pull request merged**